### ISSUE TITLE
Everything freezes when player gets power-up or shrinks back to small

### ISSUE THREAD

**Issue by vitzie629** [TheXTech/TheXTech#412]
When a player get a power-up or shrinks back to small (loses the power-up), as you can see, everything including the NPCs is moving in this animation below.
![Scr_2022-09-14_20-51-36](https://user-images.githubusercontent.com/90484844/190288260-2ab82c3a-b3df-4cb7-a794-8938745b3b7f.gif)
However, I would like to be allowed to have an option to let everything freeze when the player grabs a power-up or loses a power-up just like real Mario games when a player grows or shrinks, everything freezes. To enable the option, you would use compat.ini to enable everything frozen when player picks up or loses power-up just like real Mario games or false & let SMBX stay normal for those kind of physics.

**Comment by Wohlstand** [comment]
When singleplayer, that could work. For multiplayer this should being disabled as players shouldn't block each other by power up effects changes.

**Comment by vitzie629** [comment]
> When singleplayer, that could work. For multiplayer this should being disabled as players shouldn't block each other by power up effects changes.

Interesting! For singleplayer, that's a great idea! Multiplayer on the other hand, would remain the same as always since Redigit introduced SMBX back in 2009 with v1.0!

**Comment by Alucard648** [comment]
1. Player rides fast moving railed platform.
2. Player gets hit by a random enemy, like chasing Koopa.
3. During power-down animation platform continues to move, usually off-screen and despawn, while player is frozen in place.
4. Player is dropped into bottomless pit/lava. Instant death.

**Comment by Wohlstand** [comment]
In SMBX (and TheXTech), all moving layers and railed platform gets suspension when player takes transform effect.

**Comment by Alucard648** [comment]
What about other rideable NPCs, like skull rafts, moving tanks etc. ?

**Comment by ds-sloth** [comment]
I do not think we should implement this. TheXTech is based on SMBX64 logic, not the logic of any other game.

**Comment by LoveBodhi** [comment]
Actually "Power Up/Down Freeze"...

**Comment by LoveBodhi** [comment]
Gatete Mario Engine have "Power-up/down Freeze" but not in Hello Mario Engine.

**Comment by ChristianSilvermoon** [comment]
I am for it for playabillity reasons

The video **Stop for Big Moments!**  by *Masahiro Sakurai* does a good job showcasing why it'd be beneficial.
[![image](https://i.ytimg.com/an_webp/OdVkEOzdCPw/mqdefault_6s.webp?du=3000&sqp=CNC4mZkG&rs=AOn4CLCEB-7yRyc2VdEVn_7NG2evSRzeDA)](https://www.youtube.com/watch?v=OdVkEOzdCPw)
*The thumbnail here is actually a link to the video on YouTube*

Hit Stop should only occur in Modes 0 and 1 for the sake of compatabillity

**Comment by Wohlstand** [comment]
BTW, if attempt to make this now, that seems easy to re-use the existing "FreezeNPCs" global flag (shared with the green P-Switch).

**Comment by Wohlstand** [comment]
Do you mean, just keep this feature being an episode-side opt-in feature? So, old episodes will work as-is, but when an author makes a new episode, they could enable this feature for an episode.

**Comment by 0lhi** [comment]
*If* it's added then it should definitely be opt-in.

However, I don't think that stuff should be added just because one particular franchise has done it before. One reason for that is that some of the original implementations are objectively worse (like that SFX Combo that can cause huge irritation at Power-Up Generators), or don't integrate well here (this Issue would be limited to Singleplayer, creating an inconsistency).

The other reason I will share with you via PM. It may also be a point for this Issue to be more appropriate for Moondust than for TheXTech.

**Comment by ChristianSilvermoon** [comment]
I'd prefer player-side opt-in

This is *not* a franchise-specific feature, it is done by **many** different games.

**Comment by ds-sloth** [comment]
I can definitely see the reasoning here, and it's not without precedent for us to make new QoL features (like World Map Fast Move) that weren't included in SMBX logic.

For future reference, I want to emphasize that QoL feature requests are encouraged, but it is totally outside of the scope of our project to match any existing game franchises. I'm changing the topic title to reflect that. We have one target -- the SMBX64 engine -- and two minor sources of inspiration (SMBX-38A and SMBX2). If you want us to add a feature, explain why it is helpful on its own. It being in an unrelated game won't make us more likely to implement it.
